packages/flutter/lib/src/cupertino/radio.dart

```diff
@@ -0,0 +1,312 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/widgets.dart';
+
+import 'colors.dart';
+import 'toggleable.dart';
+
+// Examples can assume:
+// late BuildContext context;
+// enum SingingCharacter { lafayette }
+// late SingingCharacter? _character;
+// late StateSetter setState;
+
+const double _kOuterRadius = 7.0;
+const double _kInnerRadius = 2.975;
+
+// The relative values needed to transform a color to it's equivilant focus
+// outline color.
+const double _kCupertinoFocusColorOpacity = 0.80;
+const double _kCupertinoFocusColorBrightness = 0.69;
+const double _kCupertinoFocusColorSaturation = 0.835;
+
+/// A macOS style radio button.
+///
+/// Used to select between a number of mutually exclusive values. When one radio
+/// button in a group is selected, the other radio buttons in the group cease to
+/// be selected. The values are of type `T`, the type parameter of the
+/// [CupertinoRadio] class. Enums are commonly used for this purpose.
+///
+/// The radio button itself does not maintain any state. Instead, selecting the
+/// radio invokes the [onChanged] callback, passing [value] as a parameter. If
+/// [groupValue] and [value] match, this radio will be selected. Most widgets
+/// will respond to [onChanged] by calling [State.setState] to update the
+/// radio button's [groupValue].
+///
+/// {@tool dartpad}
+/// Here is an example of CupertinoRadio widgets wrapped in CupertinoListTiles.
+///
+/// The currently selected character is passed into `groupValue`, which is
+/// maintained by the example's `State`. In this case, the first [CupertinoRadio]
+/// will start off selected because `_character` is initialized to
+/// `SingingCharacter.lafayette`.
+///
+/// If the second radio button is pressed, the example's state is updated
+/// with `setState`, updating `_character` to `SingingCharacter.jefferson`.
+/// This causes the buttons to rebuild with the updated `groupValue`, and
+/// therefore the selection of the second button.
+///
+/// ** See code in examples/api/lib/cupertino/radio/radio.0.dart **
+/// {@end-tool}
+///
+/// See also:
+///
+///  * [CupertinoSlider], for selecting a value in a range.
+///  * [CupertinoCheckbox] and [CupertinoSwitch], for toggling a particular value on or off.
+///  * [Radio], the Material Design equivalent.
+///  * <https://developer.apple.com/design/human-interface-guidelines/components/selection-and-input/toggles/>
+class CupertinoRadio<T> extends StatefulWidget {
+  /// Creates a macOS-styled radio button.
+  ///
+  /// The radio button itself does not maintain any state. Instead, when the
```

Comment:
I don't think you need this paragraph. This is already explained in the class-level description. If you do want to keep it, it should probably be a macro to keep them in sync.